prover/src/gpu/metal/tests.rs

```diff
@@ -0,0 +1,233 @@
+#[cfg(test)]
+mod tests {
```

Comment:
nit: we don't need this internal module here. The `#[cfg(test)]` can be applied to the module import in `prover/src/gpu/metal/mod.rs`